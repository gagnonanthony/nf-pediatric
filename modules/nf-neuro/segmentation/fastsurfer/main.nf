process SEGMENTATION_FASTSURFER {
    tag "$meta.id"
    label 'process_single'

    container "${ 'deepmi/fastsurfer:cpu-v2.2.0' }"

    containerOptions '--entrypoint ""'

    input:
        tuple val(meta), path(anat), path(fs_license)

    output:
        tuple val(meta), path("*_fastsurfer")    , emit: fastsurferdirectory
        path "versions.yml"                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
        def prefix = task.ext.prefix ?: "${meta.id}"
        def acq3T = task.ext.acq3T ? "--3T" : ""
        def FASTSURFER_HOME = "/fastsurfer"
        def SUBJECTS_DIR = "${prefix}_fastsurfer"
    """
    mkdir ${prefix}_fastsurfer/
    $FASTSURFER_HOME/run_fastsurfer.sh  --allow_root \
                                        --sd \$(realpath ${SUBJECTS_DIR}) \
                                        --fs_license \$(realpath $fs_license) \
                                        --t1 \$(realpath ${anat}) \
                                        --sid ${prefix} \
                                        --seg_only --py python3 \
                                        ${acq3T}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        fastsurfer: \$($FASTSURFER_HOME/run_fastsurfer.sh --version)
    END_VERSIONS
    """

    stub:
        def prefix = task.ext.prefix ?: "${meta.id}"

    """
    $FASTSURFER_HOME/run_fastsurfer.sh --version

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        fastersurfer: 2.2.0+9f37d02
    END_VERSIONS
    """
}
