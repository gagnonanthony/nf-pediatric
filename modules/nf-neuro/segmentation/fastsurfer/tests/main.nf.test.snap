{
    "segmentation - fastsurfer": {
        "content": [
            "test_fastsurfer",
            [
                "versions.yml:md5,89a1cae763de31f351ea51b92c49fe41"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0-rc1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-05T14:37:22.562225"
    }
}